pkg/chains/objects/objects_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package objects
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/pod"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	namespace  = "objects-test"
+	pullSecret = "pull-secret"
+)
+
+var (
+	tr = &v1beta1.TaskRun{
```

Comment:
I think this is problematic because the pointer is shared across multiple unit tests. This can cause unintended consequences.

Can we maybe do away with these variables and just have some factory-like functions that create them as needed? e.g.

```go
...
		{
			name:     "Test pull secret found",
			taskRun:  fakeTaskRunObject(true),
			want:     []string{pullSecret},
		}
...


func fakeTaskRunObject(includePullSecret bool) *v1beta1.TaskRun {
  tr := &v1beta1.TaskRun{
	ObjectMeta: metav1.ObjectMeta{
			Name:      "foo",
			Namespace: namespace,
		},
		Spec: v1beta1.TaskRunSpec{},
	}
  
  if includePullSecret {
    tr.Spec.PodTemplate = &pod.PodTemplate{
		ImagePullSecrets: []corev1.LocalObjectReference{
			{
				Name: pullSecret,
			},
		},
	}
  }

  return tr
}

```